tests/rust/src/bin/path_open_preopen.rs

```diff
@@ -0,0 +1,153 @@
+const FIRST_PREOPEN: u32 = 3;
+
+unsafe fn path_open_preopen() {
+    let prestat = wasi::fd_prestat_get(FIRST_PREOPEN).expect("fd 3 is a preopen");
```

Comment:
WASI defines file descriptors 0, 1, and 2 for stdio, but it doesn't say that further preopens will have a sequence number: https://github.com/WebAssembly/WASI/blob/60cff3164f66d15092248a5adfa901a4caa7fea8/legacy/application-abi.md?plain=1#L54

`path_open`, even though it's not quite the same, explicitly specifies that the number will be randomized: https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/witx/wasi_snapshot_preview1.witx#L355

Reply:
Good point - I've added a function which runs through all possible values of an `fd` starting from 3 and returns the first one which is a preopen. 

I think this is fine for the moment since wasmtime and WAMR seem to open preopened fd's in sequence after the stdio ones anyway. But it might be an issue if another runtime really did choose a random number since the function would take a long time to find the first preopened fd 😄 given there are 2^31 possibilities.